File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distance = euclidean_distance({0, 0}, {x, y})
    get_score(distance)
  end

  defp get_score(distance) when distance <= 1, do: 10
  defp get_score(distance) when distance <= 5, do: 5
  defp get_score(distance) when distance <= 10, do: 1
  defp get_score(_), do: 0

  defp euclidean_distance({x1, y1}, {x2, y2}), do: :math.sqrt((x1-x2) ** 2 + (y1-y2) ** 2)
end

File: solutions/erlang/darts/1/src/darts.erl
-module(darts).

-export([score/2]).


score(X, Y) -> 
    Zone = zone({X, Y}),
    points(Zone).

zone(Position) ->
    Radius = radius(Position),
    circle(Radius).

points(inner) -> 10;
points(middle) -> 5;
points(outer) -> 1;
points(outside) -> 0.

circle(Radius) when Radius =< 1 -> inner;
circle(Radius) when Radius =< 5 -> middle;
circle(Radius) when Radius =< 10 -> outer;
circle(_) -> outside.

radius({X, Y}) -> math:sqrt(X * X + Y * Y).
